docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,205 @@
+==================
+Introduction
+==================
+
+Relay is differentiable programming language with support for
+closures, control-flow, and recursion. It has an advanced
+static type system specifically designed for programs written
+by machine  learning practioniers and researchers.
+
+Relay is intended to replace the computation graph based
+intermediate representations currently employed by deep
+learning frameworks and compilers. The deep learning community
+has organically evolved a representation that was useful
+for the form of computation originally desired, i.e
+a directed acyclic graph of primitive functions.
+
+This may be a useful way to describe early ML models, but is
+limited onceyou want to generalize to dynamic models.
+
+At the same time computation graphs have been over loaded
+purpose as both a compile-time and run-time data structure.
+The conflation of the description of a computation, the
+representation used for optimizing it, and the data structure
+used to execute it uncessarily hampers many of the goals
+of a framework.
```

Comment:
This sentence is phrased rather confusingly. Perhaps the following might be clearer?

"Unnecessarily conflating the description, representation, and execution environment of a computation hampers many of the goals of a framework." (Also appears that this may need to be explained in greater detail.)